docs/sphinxext/mantiddoc/directives/amalgamate.py

```diff
@@ -0,0 +1,67 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantiddoc.directives.base import BaseDirective
+
+import os
+
+
+class CompilationDirective(BaseDirective):
+
+    """
+    Obtains the summary for a given algorithm based on it's name.
+    """
+
+    required_arguments, optional_arguments = 1, 1
+
+    def run(self):
+        """
+        The main entry point that docutils calls.
+        It calls self.execute to do the main work.
+        Derived classes should override execute() and insert
+        whatever rst they require with self.add_rst()
+        """
+        nodes = self.execute()
+        if self.rst_lines is not None:
+            self.commit_rst()
+        return nodes
+
+    def execute(self):
+        """
+        Called by Sphinx when the ..compiler:: directive is encountered.
+        """
+        script_dir = self.getPath()
+        for file in os.listdir(script_dir):
+            if file.endswith(".rst"):
+                with open(script_dir + '/' + file) as f:
+                    contents = f.read()
+                    self.add_rst(contents)
+
+        return []
+
+    def getPath(self):
+        # gets the current directory
+        current_dir = os.getcwd()
+        # the location of documentation
+        source_dir = '../../mantid/docs/source/'
```

Comment:
This break for me, as it assumes the top level directory is called `mantid`:
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/david/mantid/docs/source/release/v6.4.0/Diffraction/Engineering/New_features'
```

(The above should be mantidpr)